File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/java/io/github/lishangbu/avalon/dataset/service/TypeService.java
package io.github.lishangbu.avalon.dataset.service;

import io.github.lishangbu.avalon.dataset.entity.Type;
import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/// 属性服务。
public interface TypeService {

    /// 根据条件分页查询属性。
    Page<Type> getPageByCondition(Type type, Pageable pageable);

    /// 新增属性。
    Type save(Type type);

    /// 更新属性。
    Type update(Type type);

    /// 根据主键删除属性。
    void removeById(Long id);

    /// 根据条件查询属性列表。
    List<Type> listByCondition(Type type);
}

File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/kotlin/io/github/lishangbu/avalon/dataset/service/TypeService.kt
package io.github.lishangbu.avalon.dataset.service

import io.github.lishangbu.avalon.dataset.entity.dto.SaveTypeInput
import io.github.lishangbu.avalon.dataset.entity.dto.TypeSpecification
import io.github.lishangbu.avalon.dataset.entity.dto.TypeView
import io.github.lishangbu.avalon.dataset.entity.dto.UpdateTypeInput

/** 属性应用服务 */
interface TypeService {
    /** 创建属性 */
    fun save(command: SaveTypeInput): TypeView

    /** 更新属性 */
    fun update(command: UpdateTypeInput): TypeView

    /** 删除指定 ID 的属性 */
    fun removeById(id: Long)

    /** 按筛选条件查询属性列表 */
    fun listByCondition(specification: TypeSpecification): List<TypeView>
}
